pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgrescluster_types.go

```diff
@@ -531,6 +531,44 @@ type PostgresVolumesSpec struct {
 	// ---
 	// +optional
 	Temp *VolumeClaimSpec `json:"temp,omitempty"`
+
+	// Additional pre-existing volumes to add to the pod.
+	// ---
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=10
+	Additional []*AdditionalVolume `json:"additional,omitempty"`
+}
+
+type AdditionalVolume struct {
+	// The name of the volume used for mounting path.
+	// Volumes are mounted in the pods at `volumes/<NAME>`
+	// Must be unique.
+	// ---
+	// The `Name` field is a `DNS1123Subdomain` type to enforce
+	// the max length and also allow us to more easily transition
+	// to CPK-provisioned volumes.
+	// +kubebuilder:validation:Required
+	Name DNS1123Subdomain `json:"name,omitempty"`
+
+	// A reference to a preexisting PVC.
+	// ---
+	// +kubebuilder:validation:Required
+	ClaimName string `json:"claimName,omitempty"`
+
+	// The containers to attach this volume to.
+	// A blank/unset `Containers` field matches all containers.
+	// ---
+	// +optional
+	// +listType=set
+	// +listType=atomic
```

Comment:
🤔 Hmm. These markers arrive in the same OpenAPI extension. I'm surprised one can't declare an "atomic set of unique values." If we can only have one, I prefer **atomic** for its effect on managed fields. Controllers can do `sets.New(s...)` easily enough.

- https://kubernetes.io/docs/reference/using-api/server-side-apply#merge-strategy
- There are some [open issues about the extension](https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue%20state%3Aopen%20%22x-kubernetes-list-type%22).

🔧 We should set a max length here, too. Every new list and map should have a max.